src/app/date-slider/date-slider.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, OnInit, Input, OnChanges, AfterViewInit, EventEmitter, Output, ViewEncapsulation } from '@angular/core';
+import { HelperService } from '../helper.service';
+import 'jquery';
+declare var $: any;
+import 'ion-rangeslider';
+
+@Component({
+  selector: 'app-date-slider',
+  templateUrl: './date-slider.component.html',
+  styleUrls: ['./date-slider.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class DateSliderComponent implements OnInit, OnChanges, AfterViewInit {
+  @Input() subCat;
+  @Input() dates;
+  @Input() freq;
+  @Input() dateFrom;
+  @Input() dateTo;
+  @Input() sublist;
+  @Output() updateRange = new EventEmitter(true);
+  private start;
+  private end;
+
+  constructor(private _helper: HelperService) { }
+
+  ngOnInit() {
+  }
+
+  ngAfterViewInit() {
+    const that = this;
+    const freq = this.freq;
+    $('#' + this.subCat.id).ionRangeSlider({   min: 0,
+      from: this.start,
+      to: this.end,
+      values: this.dates,
+      prettify_enabled: false,
+      hide_min_max: true,
+      keyboard: true,
+      keyboard_step: 1,
+      type: 'double',
+      onFinish: function(data) {
+        const chartStart = that.formatChartDate(data.from_value, freq);
+        const chartEnd = that.formatChartDate(data.to_value, freq);
+        const tableStart = data.from_value.toString();
+        const tableEnd = data.to_value.toString();
+        that.updateRange.emit({ chartStart: chartStart, chartEnd: chartEnd, tableStart: tableStart, tableEnd: tableEnd });
+      }
+    });
+  }
+
+  ngOnChanges() {
+    if (this.dates && this.dates.length) {
+      const defaultRanges = this._helper.setDefaultRange(this.freq, this.dates);
+      let startIndex = defaultRanges.start, endIndex = defaultRanges.end;
+      this.dates.forEach((date, index) => {
+        // Range slider is converting annual year strings to numbers
+        if (date == this.dateFrom) {
+          startIndex = index;
+        }
+        if (date == this.dateTo) {
+          endIndex = index;
+        }
+      });
+      // Start and end used for 'from' and 'to' inputs in slider
+      // If start/end exist in values array, position handles at start/end; otherwise, use default range
+      this.start = startIndex;
+      this.end = endIndex;
+      this.sublist.forEach((sub) => {
+        if ($('#' + sub.id).data('ionRangeSlider')) {
+          $('#' + sub.id).data('ionRangeSlider').update({
```

Comment:
I would recommend storing the jQuery request instead of running it twice.
`const slider = $('#' + sub.id).data('ionRangeSlider')`